valhalla/sif/dynamiccost.h

```diff
@@ -497,10 +497,10 @@ class DynamicCost {
 
     // If forward, check if the edge marks the end of a restriction, else check
     // if the edge marks the start of a complex restriction.
-    if ((forward && (edge->end_restriction() & access_mode())) ||
-        (!forward && (edge->start_restriction() & access_mode()))) {
+    if ((forward && (edge->end_restriction() & access_mask_)) ||
```

Comment:
this is in `Restricted` for complex restrictions. instead of doing this I'd rather do what `EvaluateRestrictions` is doing and pass the access mask as argument. before this PR that was actually fully redundant but now it does make sense to pass it as argument.